### ISSUE TITLE
Invalid cachedPointStyle

### ISSUE THREAD

**Issue by IIFelix** [NieuwlandGeo/SLDReader#132]
Suppose you have an sld with a Point- and LineSymbolizer. You then create a style for a linestring geometry. You will get back this:

```json
{
  "geometry_": {
    "disposed": false,
    "pendingRemovals_": null,
    "dispatching_": null,
    "listeners_": null,
    "revision_": 1,
    "ol_uid": "362",
    "values_": null,
    "extent_": [
      null,
      null,
      null,
      null
    ],
    "extentRevision_": -1,
    "simplifiedGeometryMaxMinSquaredTolerance": 0.07597034037581123,
    "simplifiedGeometryRevision": 1,
    "layout": "XY",
    "stride": 2,
    "flatCoordinates": [
      500641.7758006881,
      5392759.1374269985
    ]
  },
  "fill_": null,
  "image_": {
    "opacity_": 1,
    "rotateWithView_": false,
    "rotation_": 0,
    "scale_": 1,
    "scaleArray_": [
      1,
      1
    ],
    "displacement_": [
      0,
      0
    ],
    "canvas_": {
      "1": {}
    },
    "hitDetectionCanvas_": {},
    "fill_": {
      "color_": "rgba(199, 21, 133, 0)"
    },
    "origin_": [
      0,
      0
    ],
    "points_": null,
    "radius_": 9,
    "angle_": 0,
    "stroke_": {
      "color_": "rgba(199, 21, 133, 0.5)",
      "lineDash_": null,
      "lineDashOffset_": null,
      "width_": 18
    },
    "size_": [
      36,
      36
    ],
    "renderOptions_": {
      "strokeStyle": "rgba(199, 21, 133, 0.5)",
      "strokeWidth": 18,
      "size": 36,
      "lineDash": null,
      "lineDashOffset": null,
      "lineJoin": "round",
      "miterLimit": 10
    }
  },
  "renderer_": null,
  "hitDetectionRenderer_": null,
  "stroke_": null,
  "text_": null
}
```

The "geometry" property is important in the json above.

Now if you create a new style for a point geometry the memoizeStyleFunction will give you back the same style created by the linestring geometry because it has the same symbolizer. But the "geometry" in the cached style will not have the coordinates of the point geometry. Instead it has the coordinates calculated by the "getLineMidpoint" function of the linestring.

As a result openlayers will not render the coordinates of the point geometry but rather the coordinates in the "geometry" property of the style.

**Comment by ajkopinga** [comment]
Thanks for finding this bug! It's a sneaky one. 🥷 

I've fixed it by resetting the style geometry to null when a point style is generated for a point feature. This gets rid of any previous calculated geometry.

I'll release a new library version soon.

**Comment by ajkopinga** [comment]
I've released v0.2.16 that should fix this bug.

**Comment by IIFelix** [comment]
Wow that was fast. Tyvm !
